src/engine/guidance/post_processing.cpp

```diff
@@ -287,20 +347,152 @@ std::vector<RouteStep> postProcess(std::vector<RouteStep> steps)
         detail::fixFinalRoundabout(steps);
     }
 
-    // finally clean up the post-processed instructions.
-    // Remove all invalid instructions from the set of instructions.
-    // An instruction is invalid, if its NO_TURN and has WaypointType::None.
-    // Two valid NO_TURNs exist in each leg in the form of Depart/Arrive
+    return removeNoTurnInstructions(std::move(steps));
+}
 
-    // keep valid instructions
-    const auto not_is_valid = [](const RouteStep &step) {
-        return step.maneuver.instruction == TurnInstruction::NO_TURN() &&
-               step.maneuver.waypoint_type == WaypointType::None;
+// Post Processing to collapse unnecessary sets of combined instructions into a single one
+std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps)
+{
+    // invalidate a step and set its content to nothing
+    const auto invalidateStepAtIndex = [&steps](std::size_t index) {
+        auto &step = steps[index];
+        step.maneuver.instruction = TurnInstruction::NO_TURN();
+        step.duration = 0;
+        step.distance = 0;
+        step.geometry_begin = 0;
+        step.geometry_end = 0;
+        step.name.clear();
```

Comment:
Hm is this the same as swapping step with an empty one? As in `step = RouteStep();`